File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/fishlo/program/Teleop.java
package org.firstinspires.ftc.teamcode.fishlo.program;

import com.qualcomm.robotcore.eventloop.opmode.TeleOp;

import org.firstinspires.ftc.teamcode.fishlo.robot.Fishlo;
import org.firstinspires.ftc.teamcode.opMode.DriverControlledProgram;
import org.firstinspires.ftc.teamcode.robot.Robot;



@TeleOp()
public class Teleop extends DriverControlledProgram {
    private Fishlo fishlo;

    @Override
    protected Robot buildRobot() {
        fishlo = new Fishlo(this);

        return fishlo;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/TeleOp.kt
package org.firstinspires.ftc.teamcode.opmodes

import com.arcrobotics.ftclib.gamepad.GamepadEx
import com.qualcomm.robotcore.eventloop.opmode.TeleOp
import com.qualcomm.robotcore.eventloop.opmode.OpMode
import org.firstinspires.ftc.teamcode.util.RobotHardware
import org.firstinspires.ftc.teamcode.util.TelemetryLogger

lateinit var logger: TelemetryLogger

@TeleOp(name = "TeleOp")
@Suppress("unused")
class TeleOp : OpMode() {
    private lateinit var gamepad: GamepadEx
    private lateinit var hardware: RobotHardware

    override fun init() {
        gamepad = GamepadEx(gamepad1)
        logger = TelemetryLogger(telemetry)

        logger.info("Initialized OpMode")
    }

    override fun start() {
        hardware = RobotHardware(hardwareMap, gamepad)
        logger.info("Started OpMode")
    }

    override fun loop() {
        hardware.driveTrain.drive()
        gamepad.readButtons()
    }
}